torch/_dynamo/variables/constant.py

```diff
@@ -148,6 +153,14 @@ def has_arith_binop(num_ty):
             search = args[0].as_python_constant()
             result = search in self.value
             return ConstantVariable(result, **options)
+        elif name == "append":
+            assert not kwargs
+            assert len(args) == 1
+            arg = args[0]
+            new_value = self.value.append(arg.as_python_constant)
```

Comment:
not method call?

Reply:
Mutating self.value as seen here seems wrong... variable trackers are immutable

Reply:
This is a common pattern because of the `tx.replace_all(self, new_obj)` below

Reply:
But what if the variable tracker is stored in a checkpoint? You've accidentally mutated the checkpoint state too!